posthog/exception_capture.py

```diff
@@ -0,0 +1,54 @@
+import json
+import sys
+import threading
+from typing import TYPE_CHECKING
+
+from posthog.exception_utils import single_exception_from_error_tuple
+
+if TYPE_CHECKING:
+    from posthog.client import Client
+
+
+class ExceptionCapture:
+    # TODO: Some method of avoiding internal SDK exceptions?
```

Comment:
the SDK has to degrade gracefully which means wrapping up the whole SDK with a try-catch and fallback to logging if something goes wrong, but it should never capture its exception otherwise it causes an infinite loop